### PR TITLE
stages/ostree.deploy.container: add imgref regex validation

### PR DESCRIPTION
**Pull request by lukewarmtemp**
This adds a regex condition for the type of container image validation that is accepted. Examples target image refs are also included.

### CODE REVIEW

File: stages/org.osbuild.ostree.deploy.container [hunk 1]
```diff
@@ -41,7 +41,11 @@ SCHEMA_2 = """
     },
     "target_imgref": {
       "description": "imageref used as the source of truth for updates",
-      "type": "string"
+      "type": "string",
+      "pattern": "^(ostree-remote-registry|ostree-image-signed|ostree-unverified-registry):.*$",
```
[the remark this answers is not in the document]
Comment (lukewarmtemp) [reply stages/org.osbuild.ostree.deploy.container:45]: Hm, I'm not too sure. I can dig into the `ostree-rs-ext` repo to find all the options listed and test them out though.

[the remark this answers is not in the document]
Comment (lukewarmtemp) [reply stages/org.osbuild.ostree.deploy.container:45]: From the ostree native container docs: https://coreos.github.io/rpm-ostree/container/#url-format-for-ostree-native-containers, we have the following options:
```
ostree-unverified-image:registry:<oci image>
```

[the remark this answers is not in the document]
Comment (lukewarmtemp) [reply stages/org.osbuild.ostree.deploy.container:45]: From the ostree native container docs: https://coreos.github.io/rpm-ostree/container/#url-format-for-ostree-native-containers, we have the following options:

1. `ostree-unverified-registry:<oci image>`
    - shorthand for `ostree-unverified-image:registry:<oci image>`
  
2. `ostree-remote-registry:<ostree remote>:<oci image>`
    - shorthand for `ostree-remote-image:<ostree remote>:registry:<oci image>`
    - shorthand for `ostree-remote-image:<ostree remote>:docker://<oci image>`
  
3. `ostree-image-signed:registry:<oci image>` or `ostree-image-signed:docker://<oci image>`

Just based off this documentation, all options fall under `ostree-unverified-registry`,  `ostree-remote-registry`, and `ostree-image-signed`

[the remark this answers is not in the document]
Comment (dustymabe) [reply stages/org.osbuild.ostree.deploy.container:45]: sounds good. thanks!

**Comment by dustymabe** [comment]
squashed/merged into https://github.com/osbuild/osbuild/pull/1402

**Pull request closed without merging**